bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TypedescValueImpl.java

```diff
@@ -91,6 +93,9 @@ public Object instantiate(Strand s, BInitialValueEntry[] initialValues) {
             return new MapValueImpl(this.describingType, (BMapInitialValueEntry[]) initialValues);
         } else if (referredType.getTag() == TypeTags.TUPLE_TAG) {
             return new TupleValueImpl(this.describingType, (BListInitialValueEntry[]) initialValues);
+        } else if (getEffectiveType(referredType).getTag() == TypeTags.TUPLE_TAG) {
```

Comment:
Shall we store `getEffectiveType(referredType)` result in a local variable and reuse it instead of calling the method twice?